.github/workflows/im-reusable-setup-deployment-workflow.yml

```diff
@@ -0,0 +1,100 @@
+# The purpose of this workflow is to verify that the branch/tag/sha to deploy is a valid ref
+# and for production deploys to verify that the tag is reachable from the default branch
+# (meaning it has been reviewed and merged into main) and that the release is production
+# ready (it is not a draft/pre-release).
+
+
+# Example Usage in a repo's workflow:
+# jobs:
+#  setup-deployment-workflow:
+#    uses: im-practices/.github/.github/workflows/im-reusable-setup-deployment-workflow@v2
+#    with:
+#      ref-to-deploy: v1.2.3
+#      deployment-environment: prod
+#      production-environments: 'prod,prod-secondary'
+#      default-branch: main
+
+on:
+  workflow_call:
+    inputs:
+      ref-to-deploy:
+        description: The branch, tag or sha that will be deployed.
+        type: string
+        required: true
+      deployment-environment:
+        description: 'The environment that is being deployed to: dev, qa, stage, stage-secondary, uat, demo, prod, prod-secondary.'
+        required: true
+        type: string
+      production-environments:
+        description: Comma separated list of production environments to check against.  Defaults to 'prod,prod-secondary'
+        required: false
+        type: string
+        default: 'prod,prod-secondary'      
```

Comment:
Should `demo` be considered a production env. as well?

Reply:
I'm not sure about that environment.  I would think if the team wants to include demo they can override the default here.